tests/components/opentherm_gw/test_init.py

```diff
@@ -80,5 +87,65 @@ async def test_device_registry_update(
         await setup.async_setup_component(hass, DOMAIN, {})
 
     await hass.async_block_till_done()
-    gw_dev = device_registry.async_get_device(identifiers={(DOMAIN, MOCK_GATEWAY_ID)})
+    gw_dev = device_registry.async_get_device(
+        identifiers={(DOMAIN, f"{MOCK_GATEWAY_ID}-{OpenThermDeviceIdentifier.GATEWAY}")}
+    )
+    assert gw_dev is not None
     assert gw_dev.sw_version == VERSION_NEW
+
+
+# Device migration test can be removed in 2025.4.0
+async def test_device_migration(
+    hass: HomeAssistant, device_registry: dr.DeviceRegistry
+) -> None:
+    """Test that the device registry is updated correctly."""
+    MOCK_CONFIG_ENTRY.add_to_hass(hass)
+
+    device_registry.async_get_or_create(
+        config_entry_id=MOCK_CONFIG_ENTRY.entry_id,
+        identifiers={
+            (DOMAIN, MOCK_GATEWAY_ID),
+        },
+        name="Mock Gateway",
+        manufacturer="Schelte Bron",
+        model="OpenTherm Gateway",
+        sw_version=VERSION_OLD,
+    )
+
+    with (
+        patch("pyotgw.OpenThermGateway.set_control_setpoint"),
+        patch("pyotgw.OpenThermGateway.set_max_relative_mod"),
+        patch("pyotgw.OpenThermGateway.disconnect"),
+        patch("pyotgw.OpenThermGateway.connect", return_value=MINIMAL_STATUS_UPD),
```

Comment:
We should patch the library where we use it. I recommend patching the full object:

`homeassistant.components.opentherm_gw.OpenThermGateway`

```py
with patch("homeassistant.components.opentherm_gw.OpenThermGateway") as gateway:
    gateway.return_value.connect.return_value = MINIMAL_STATUS_UPD
```